etc/spack/defaults/concretizer.yaml

```diff
@@ -28,3 +28,9 @@ concretizer:
     # instance concretize with target "icelake" while running on "haswell").
     # If "true" only allow targets that are compatible with the host.
     host_compatible: true
+  # When "true" concretize root specs of environments together, so that each unique
+  # package in an environment corresponds to one concrete spec. This ensures
```

Comment:
Future work, writing about it here since this comment made me think of it -- we should add a feature that this setting is applied to specs from the command line when multiple specs are provided (e.g. `spack install foo bar baz` could co-concretize).